ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/command/toolbar/selector/PanelSelectorViewImpl.java

```diff
@@ -0,0 +1,183 @@
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import elemental.dom.Element;
+import elemental.dom.Node;
+import elemental.events.Event;
+import elemental.events.EventListener;
+import elemental.events.EventRemover;
+import org.eclipse.che.ide.Resources;
+import org.eclipse.che.ide.util.dom.Elements;
+import org.vectomatic.dom.svg.ui.SVGImage;
+import org.vectomatic.dom.svg.ui.SVGResource;
+
+/**
+ * Implementation for {PanelSelectorView}.
+ */
+@Singleton
+public class PanelSelectorViewImpl extends Composite implements PanelSelectorView {
+
+    interface PanelSelectorViewImplUiBinder extends UiBinder<Widget, PanelSelectorViewImpl> {
+    }
+
+    private ActionDelegate delegate;
+
+    private Resources resources;
+
+    @UiField
+    FlowPanel button;
+
+    @UiField
+    FlowPanel popup;
+
+    @UiField
+    FlowPanel selectorLeft;
+
+    @UiField
+    FlowPanel selectorLeftBottom;
+
+    @UiField
+    FlowPanel selectorFullEditor;
+
+    @UiField
+    FlowPanel selectorBottom;
+
+    @UiField
+    FlowPanel selectorRight;
+
+    @UiField
+    FlowPanel selectorLeftRightBottom;
+
+    @UiField
+    FlowPanel iconPanel;
+
+    private EventRemover clickListenerRemover;
+
+    @Inject
+    public PanelSelectorViewImpl(PanelSelectorViewImplUiBinder uiBinder,
+                                 Resources resources) {
+        this.resources = resources;
+        initWidget(uiBinder.createAndBindUi(this));
+
+        button.addDomHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent clickEvent) {
+                delegate.onButtonClicked();
+            }
+        }, ClickEvent.getType());
+
+    }
+
+    @Override
+    public void setDelegate(ActionDelegate delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void setState(State state) {
+        switch (state) {
+            case LEFT:
+                setIcon(resources.panelSelectorLeft());
+                break;
+
+            case LEFT_BOTTOM:
+                setIcon(resources.panelSelectorLeftBottom());
+                break;
+
+            case FULL_EDITOR:
+                setIcon(resources.panelSelectorFullEditor());
+                break;
+
+            case BOTTOM:
+                setIcon(resources.panelSelectorBottom());
+                break;
+
+            case RIGHT:
+                setIcon(resources.panelSelectorRight());
+                break;
+
+            case LEFT_RIGHT_BOTTOM:
+                setIcon(resources.panelSelectorLeftRightBottom());
+                break;
+        }
+    }
+
+    private void setIcon(SVGResource resource) {
+        iconPanel.clear();
+        SVGImage image = new SVGImage(resource);
+        iconPanel.add(image);
+    }
+
+    @Override
+    public void showPopup() {
+        if ("block".equals(popup.getElement().getStyle().getProperty("display"))) {
```

Comment:
You're operating with properties directly instead of using specialized methods for that.